raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2849,14 +2858,15 @@ private long maybeSendRequest(
         long currentTimeMs,
         Set<ReplicaKey> remoteVoters,
         Function<Integer, Node> destinationSupplier,
-        Function<ReplicaKey, ApiMessage> requestSupplier
+        Function<ReplicaKey, ApiMessage> requestSupplier,
+        ApiKeys requestType
```

Comment:
Don't you want this type?
```java
        Function<ReplicaKey, RequestSupplier> requestSupplier,
```

This may fix some of the other code where you can't directly call `build.*Request`.